go/vt/vtgateproxy/vtgateproxy.go

```diff
@@ -170,7 +170,10 @@ func (proxy *VTGateProxy) Execute(ctx context.Context, session *vtgateconn.VTGat
 
 	// Intercept "use" statements since they just have to update the local session
 	if strings.HasPrefix(sql, "use ") {
-		targetString := sqlescape.UnescapeID(sql[4:])
+		targetString, err := sqlescape.UnescapeID(sql[4:])
+		if err != nil {
```

Comment:
Somewhat minor question but actually a bit thought provoking as well...

Suppose this parsing fails for whatever reason. Is it better to fail (as you did here) or just fall through and call `session.Execute` below? Since the whole point of this `use` statement hijacking is to avoid a round trip, if for some rare reason vtgate is able to process it but the proxy isn't then could we be unnecesarily failing?

Then again maybe the case where we can't unescape an ID is rare enough that it's just plain irrelevant?

@henryr curious about your thoughts as well.

Reply:
I think it's ok to fail. I think it's just as likely that we create a dependency elsewhere in the proxy where we assume that we've always successfully intercepted `use` statements, as a situation arises where we can't unescape the argument.